From 46cdcc1e1e825e0e7783b9587fa664f6abae58bd Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 12 Sep 2019 21:22:44 +0200
Subject: [PATCH 1/8] moves the indicator from 1/3 to the middle of index in
 pager view

---
 pager.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pager.c b/pager.c
index 28728d0f..a1857010 100644
--- a/pager.c
+++ b/pager.c
@@ -1689,7 +1689,7 @@ static void pager_menu_redraw (MUTTMENU *pager_menu)
     else
       rd->indexlen = PagerIndexLines;
 
-    rd->indicator = rd->indexlen / 3;
+    rd->indicator = rd->indexlen / 2 - 1;
 
     memcpy (rd->pager_window, MuttIndexWindow, sizeof(mutt_window_t));
     memcpy (rd->pager_status_window, MuttStatusWindow, sizeof(mutt_window_t));
@@ -1968,7 +1968,7 @@ mutt_pager (const char *banner, const char *fname, int flags, pager_t *extra)
   rd.flags = flags;
   rd.extra = extra;
   rd.indexlen = PagerIndexLines;
-  rd.indicator = rd.indexlen / 3;
+  rd.indicator = rd.indexlen / 2 - 1;
   rd.searchbuf = searchbuf;
   rd.has_types = (IsHeader(extra) || (flags & MUTT_SHOWCOLOR)) ? MUTT_TYPES : 0; /* main message or rfc822 attachment */
 
-- 
2.29.2

