From 594f24e52f6f78c39fbb172708791f2b09c0636a Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 12 Sep 2019 21:32:28 +0200
Subject: [PATCH 2/8] shows message as read as soon as it is opened

---
 hdrline.c | 5 ++---
 mutt.h    | 1 -
 mx.c      | 1 -
 pager.c   | 3 ---
 4 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/hdrline.c b/hdrline.c
index 797d8946..eb441b02 100644
--- a/hdrline.c
+++ b/hdrline.c
@@ -619,7 +619,7 @@ hdr_format_str (char *dest,
 	ch = '!';
       else if (hdr->replied)
 	ch = 'r';
-      else if (hdr->read && (ctx && ctx->msgnotreadyet != hdr->msgno))
+      else if (hdr->read)
 	ch = '-';
       else if (hdr->old)
 	ch = 'O';
@@ -696,8 +696,7 @@ hdr_format_str (char *dest,
 
       snprintf (buf2, sizeof (buf2),
 		"%c%c%s", (THREAD_NEW ? 'n' : (THREAD_OLD ? 'o' :
-		((hdr->read && (ctx && ctx->msgnotreadyet != hdr->msgno))
-		? (hdr->replied ? 'r' : ' ') : (hdr->old ? 'O' : 'N')))),
+		(hdr->read ? (hdr->replied ? 'r' : ' ') : (hdr->old ? 'O' : 'N')))),
 		hdr->deleted ? 'D' : (hdr->attach_del ? 'd' : ch),
 		hdr->tagged ? "*" :
 		(hdr->flagged ? "!" :
diff --git a/mutt.h b/mutt.h
index bfa884f4..340ffe8a 100644
--- a/mutt.h
+++ b/mutt.h
@@ -1100,7 +1100,6 @@ typedef struct _context
   int unread;			/* how many unread messages? */
   int deleted;			/* how many deleted messages */
   int flagged;			/* how many flagged messages */
-  int msgnotreadyet;		/* which msg "new" in pager, -1 if none */
 
   short magic;			/* mailbox type */
 
diff --git a/mx.c b/mx.c
index 45f557cc..ffc21b0e 100644
--- a/mx.c
+++ b/mx.c
@@ -600,7 +600,6 @@ CONTEXT *mx_open_mailbox (const char *path, int flags, CONTEXT *pctx)
   else
     ctx->realpath = safe_strdup (realpathbuf);
 
-  ctx->msgnotreadyet = -1;
   ctx->collapsed = 0;
 
   for (rc=0; rc < RIGHTSMAX; rc++)
diff --git a/pager.c b/pager.c
index a1857010..0fe06b17 100644
--- a/pager.c
+++ b/pager.c
@@ -1990,7 +1990,6 @@ mutt_pager (const char *banner, const char *fname, int flags, pager_t *extra)
 
   if (IsHeader (extra) && !extra->hdr->read)
   {
-    Context->msgnotreadyet = extra->hdr->msgno;
     mutt_set_flag (Context, extra->hdr, MUTT_READ, 1);
   }
 
@@ -2861,7 +2860,6 @@ search_next:
 	else if (!first)
 	  mutt_set_flag (Context, extra->hdr, MUTT_READ, 1);
 	first = 0;
-        Context->msgnotreadyet = -1;
 	pager_menu->redraw |= REDRAW_STATUS | REDRAW_INDEX;
 	if (option (OPTRESOLVE))
 	{
@@ -3014,7 +3012,6 @@ search_next:
   safe_fclose (&rd.fp);
   if (IsHeader (extra))
   {
-    Context->msgnotreadyet = -1;
     switch (rc)
     {
       case -1:
-- 
2.29.2

