From 9dc8f6b753723b4a3b5eccb941efbf8b2b0b61cb Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 12 Sep 2019 21:36:56 +0200
Subject: [PATCH 4/9] don't leave pager if prev/next-entry on first/last
 message

---
 curs_main.c | 23 ++++++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/curs_main.c b/curs_main.c
index fee0a34c..4343a2eb 100644
--- a/curs_main.c
+++ b/curs_main.c
@@ -1631,6 +1631,11 @@ int mutt_index_menu (void)
 	{
 	  if (!in_pager)
 	    mutt_error _("You are on the last message.");
+          else
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
 	  break;
 	}
 	if ((menu->current = ci_next_undeleted (menu->current)) == -1)
@@ -1656,6 +1661,11 @@ int mutt_index_menu (void)
 	{
 	  if (!in_pager)
 	    mutt_error _("You are on the last message.");
+          else if (menu->menu == MENU_PAGER)
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
 	  break;
 	}
 	menu->current++;
@@ -1675,6 +1685,11 @@ int mutt_index_menu (void)
 	if (menu->current < 1)
 	{
 	  mutt_error _("You are on the first message.");
+          if (menu->menu == MENU_PAGER)
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
 	  break;
 	}
 	if ((menu->current = ci_previous_undeleted (menu->current)) == -1)
@@ -1698,7 +1713,13 @@ int mutt_index_menu (void)
         CHECK_VISIBLE;
 	if (menu->current < 1)
 	{
-	  if (!in_pager) mutt_error _("You are on the first message.");
+	  if (!in_pager)
+            mutt_error _("You are on the first message.");
+          else
+          {
+            op = OP_DISPLAY_MESSAGE;
+            continue;
+          }
 	  break;
 	}
 	menu->current--;
-- 
2.24.3 (Apple Git-128)

