From f44dbc427d4fdb3e29cca2c834ac05be1aa5811a Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 12 Sep 2019 21:37:42 +0200
Subject: [PATCH 5/8] adds subject color to thread tree symbols

---
 menu.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/menu.c b/menu.c
index a7fff292..c80af9c3 100644
--- a/menu.c
+++ b/menu.c
@@ -41,23 +41,18 @@ static void print_enriched_string (int base_color, unsigned char *s, int use_ind
   size_t k;
   size_t n = mutt_strlen ((char *)s);
   mbstate_t mbstate;
-  int tree_color;
 
   if (option (OPTCURSOROVERLAY))
   {
-    tree_color = mutt_merge_colors (base_color, ColorDefs[MT_COLOR_TREE]);
     if (use_indicator)
     {
-      tree_color = mutt_merge_colors (tree_color, ColorDefs[MT_COLOR_INDICATOR]);
       base_color = mutt_merge_colors (base_color, ColorDefs[MT_COLOR_INDICATOR]);
     }
   }
   else
   {
-    tree_color = ColorDefs[MT_COLOR_TREE];
     if (use_indicator)
     {
-      tree_color = ColorDefs[MT_COLOR_INDICATOR];
       base_color = ColorDefs[MT_COLOR_INDICATOR];
     }
   }
@@ -69,7 +64,6 @@ static void print_enriched_string (int base_color, unsigned char *s, int use_ind
   {
     if (*s < MUTT_TREE_MAX)
     {
-      ATTRSET (tree_color);
       while (*s && *s < MUTT_TREE_MAX)
       {
 	switch (*s)
@@ -186,7 +180,6 @@ static void print_enriched_string (int base_color, unsigned char *s, int use_ind
 	}
 	s++, n--;
       }
-      ATTRSET(base_color);
     }
     else if ((k = mbrtowc (&wc, (char *)s, n, &mbstate)) > 0)
     {
-- 
2.29.2

