--- x.c.old	2017-10-22 21:08:50.000000000 +0200
+++ x.c	2017-10-22 21:11:50.000000000 +0200
@@ -1547,19 +1547,20 @@
 	/* 3. composed string from input method */
 	if (len == 0)
 		return;
-	if (len == 1 && e->state & Mod1Mask) {
-		if (IS_SET(MODE_8BIT)) {
-			if (*buf < 0177) {
-				c = *buf | 0x80;
-				len = utf8encode(c, buf);
-			}
-		} else {
-			buf[1] = buf[0];
-			buf[0] = '\033';
-			len = 2;
-		}
-	}
-	ttysend(buf, len);
+  /* put utf8 on Mod3Mask instead of Mod1Mask */
+  if (len == 1){
+    if (e->state & Mod1Mask) {
+      buf[1] = buf[0];
+      buf[0] = '\033';
+      len = 2;
+    } else if (IS_SET(MODE_8BIT) && e->state & Mod3Mask) {
+      if (*buf < 0177) {
+        c = *buf | 0x80;
+        len = utf8encode(c, buf);
+      }
+    }
+  }
+  ttysend(buf, len);
 }
 
 
