From 4c0ec54f2c480e80ca595cc6b8a8b994a476a2f5 Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 17 May 2018 19:47:22 +0200
Subject: [PATCH 2/5] show message as read once it is opened

---
 hdrline.c | 6 +++---
 pager.c   | 4 ----
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/hdrline.c b/hdrline.c
index 399f3170b..5a2a60c1f 100644
--- a/hdrline.c
+++ b/hdrline.c
@@ -1126,7 +1126,7 @@ static const char *index_format_str(char *buf, size_t buflen, size_t col, int co
         wch = get_nth_wchar(FlagChars, FlagCharImportant);
       else if (hdr->replied)
         wch = get_nth_wchar(FlagChars, FlagCharReplied);
-      else if (hdr->read && (ctx && ctx->msgnotreadyet != hdr->msgno))
+      else if (hdr->read)
         wch = get_nth_wchar(FlagChars, FlagCharSEmpty);
       else if (hdr->old)
         wch = get_nth_wchar(FlagChars, FlagCharOld);
@@ -1280,7 +1280,7 @@ static const char *index_format_str(char *buf, size_t buflen, size_t col, int co
           ch = get_nth_wchar(FlagChars, FlagCharNewThread);
         else if (threads && thread_is_old(ctx, hdr))
           ch = get_nth_wchar(FlagChars, FlagCharOldThread);
-        else if (hdr->read && (ctx && (ctx->msgnotreadyet != hdr->msgno)))
+        else if (hdr->read)
         {
           if (hdr->replied)
             ch = get_nth_wchar(FlagChars, FlagCharReplied);
@@ -1347,7 +1347,7 @@ static const char *index_format_str(char *buf, size_t buflen, size_t col, int co
         first = get_nth_wchar(FlagChars, FlagCharNewThread);
       else if (threads && thread_is_old(ctx, hdr))
         first = get_nth_wchar(FlagChars, FlagCharOldThread);
-      else if (hdr->read && (ctx && (ctx->msgnotreadyet != hdr->msgno)))
+      else if (hdr->read)
       {
         if (hdr->replied)
           first = get_nth_wchar(FlagChars, FlagCharReplied);
diff --git a/pager.c b/pager.c
index 309a439da..8c2a79b45 100644
--- a/pager.c
+++ b/pager.c
@@ -2251,7 +2251,6 @@ int mutt_pager(const char *banner, const char *fname, int flags, struct Pager *e
 
   if (Context && IsHeader(extra) && !extra->hdr->read)
   {
-    Context->msgnotreadyet = extra->hdr->msgno;
     mutt_set_flag(Context, extra->hdr, MUTT_READ, 1);
   }
 
@@ -3276,7 +3275,6 @@ int mutt_pager(const char *banner, const char *fname, int flags, struct Pager *e
         else if (!first)
           mutt_set_flag(Context, extra->hdr, MUTT_READ, 1);
         first = 0;
-        Context->msgnotreadyet = -1;
         pager_menu->redraw |= REDRAW_STATUS | REDRAW_INDEX;
         if (Resolve)
         {
@@ -3431,8 +3429,6 @@ int mutt_pager(const char *banner, const char *fname, int flags, struct Pager *e
   mutt_file_fclose(&rd.fp);
   if (IsHeader(extra))
   {
-    if (Context)
-      Context->msgnotreadyet = -1;
     switch (rc)
     {
       case -1:
-- 
2.15.2 (Apple Git-101.1)

