From 76acca1ebdf847b9dbdc5f8f79dee0e0110bc9c7 Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Thu, 17 May 2018 19:50:16 +0200
Subject: [PATCH 3/5] sync with mailbox and execute mbsync

---
 curs_main.c | 14 ++++++++++++++
 functions.h |  2 ++
 opcodes.h   |  1 +
 3 files changed, 17 insertions(+)

diff --git a/curs_main.c b/curs_main.c
index 720f6ad4f..edb824d9b 100644
--- a/curs_main.c
+++ b/curs_main.c
@@ -1817,6 +1817,7 @@ int mutt_index_menu(void)
 #endif
 
       case OP_MAIN_SYNC_FOLDER:
+      case OP_MAIN_SYNC_FOLDER_MAILBOX:
 
         if (Context && !Context->mailbox->msg_count)
           break;
@@ -1880,6 +1881,19 @@ int mutt_index_menu(void)
         }
         else
           menu->redraw = REDRAW_FULL;
+
+        if (op == OP_MAIN_SYNC_FOLDER_MAILBOX)
+        {
+                mutt_endwin();
+                if (mutt_system("mbsync -a") != 0)
+                {
+                        mutt_error(_("Failed to perform external sync!"));
+                }
+                mutt_any_key_to_continue(NULL);
+                mutt_resize_screen();
+                keypad(stdscr, true);
+                clearok(stdscr, true);
+        }
         break;
 
       case OP_MAIN_QUASI_DELETE:
diff --git a/functions.h b/functions.h
index dcdcea96a..48b403135 100644
--- a/functions.h
+++ b/functions.h
@@ -219,6 +219,7 @@ const struct Binding OpMain[] = { /* map: index */
   { "sort-mailbox",              OP_SORT,                           "o" },
   { "sort-reverse",              OP_SORT_REVERSE,                   "O" },
   { "sync-mailbox",              OP_MAIN_SYNC_FOLDER,               "$" },
+  { "sync-folder-mailbox",       OP_MAIN_SYNC_FOLDER_MAILBOX,       NULL },
   { "tag-pattern",               OP_MAIN_TAG_PATTERN,               "T" },
   { "tag-subthread",             OP_TAG_SUBTHREAD,                  NULL },
   { "tag-thread",                OP_TAG_THREAD,                     "\033t" },
@@ -373,6 +374,7 @@ const struct Binding OpPager[] = { /* map: pager */
   { "sort-mailbox",              OP_SORT,                         "o" },
   { "sort-reverse",              OP_SORT_REVERSE,                 "O" },
   { "sync-mailbox",              OP_MAIN_SYNC_FOLDER,             "$" },
+  { "sync-folder-mailbox",       OP_MAIN_SYNC_FOLDER_MAILBOX,     NULL },
   { "tag-message",               OP_TAG,                          "t" },
   { "toggle-quoted",             OP_PAGER_HIDE_QUOTED,            "T" },
   { "top",                       OP_PAGER_TOP,                    "^" },
diff --git a/opcodes.h b/opcodes.h
index 3ad260f30..1afba4fd6 100644
--- a/opcodes.h
+++ b/opcodes.h
@@ -185,6 +185,7 @@
   _fmt(OP_MAIN_SET_FLAG,                  N_("set a status flag on a message")) \
   _fmt(OP_MAIN_SHOW_LIMIT,                N_("show currently active limit pattern")) \
   _fmt(OP_MAIN_SYNC_FOLDER,               N_("save changes to mailbox")) \
+  _fmt(OP_MAIN_SYNC_FOLDER_MAILBOX,       N_("save changes to maildir and synch")) \
   _fmt(OP_MAIN_TAG_PATTERN,               N_("tag messages matching a pattern")) \
   _fmt(OP_MAIN_UNDELETE_PATTERN,          N_("undelete messages matching a pattern")) \
   _fmt(OP_MAIN_UNTAG_PATTERN,             N_("untag messages matching a pattern")) \
-- 
2.15.2 (Apple Git-101.1)

