From 66e5298ef02f6f8ed7a995f2c5e04874489b255f Mon Sep 17 00:00:00 2001
From: Daniel Tameling <tamelingdaniel@gmail.com>
Date: Sun, 20 May 2018 16:59:41 +0200
Subject: [PATCH 4/5] fixed bug that mbsync wasn't executed from pager

---
 curs_main.c | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/curs_main.c b/curs_main.c
index edb824d9b..1e44ab617 100644
--- a/curs_main.c
+++ b/curs_main.c
@@ -1873,15 +1873,6 @@ int mutt_index_menu(void)
         if (!Context->mailbox->path)
           mutt_context_free(&Context);
 
-        /* if we were in the pager, redisplay the message */
-        if (menu->menu == MENU_PAGER)
-        {
-          op = OP_DISPLAY_MESSAGE;
-          continue;
-        }
-        else
-          menu->redraw = REDRAW_FULL;
-
         if (op == OP_MAIN_SYNC_FOLDER_MAILBOX)
         {
                 mutt_endwin();
@@ -1894,6 +1885,16 @@ int mutt_index_menu(void)
                 keypad(stdscr, true);
                 clearok(stdscr, true);
         }
+
+        /* if we were in the pager, redisplay the message */
+        if (menu->menu == MENU_PAGER)
+        {
+          op = OP_DISPLAY_MESSAGE;
+          continue;
+        }
+        else
+          menu->redraw = REDRAW_FULL;
+
         break;
 
       case OP_MAIN_QUASI_DELETE:
-- 
2.15.2 (Apple Git-101.1)

